src/shared/loop-util.c

```diff
@@ -928,12 +978,29 @@ int loop_device_refresh_size(LoopDevice *d, uint64_t offset, uint64_t size) {
 }
 
 int loop_device_flock(LoopDevice *d, int operation) {
+        assert(IN_SET(operation & ~LOCK_NB, LOCK_UN, LOCK_SH, LOCK_EX));
         assert(d);
 
-        if (d->fd < 0)
-                return -EBADF;
+        /* When unlocking just close the lock fd */
+        if ((operation & ~LOCK_NB) == LOCK_UN) {
+                d->lock_fd = safe_close(d->lock_fd);
+                return 0;
+        }
+
+        /* If we had no lock fd so far, create one and lock it right-away */
+        if (d->lock_fd < 0) {
+                if (d->fd < 0)
+                        return -EBADF;
```

Comment:
Shouldn't we just assert here? Calling this function with a loop device without an fd seems like a programming error